tiledb/sm/query/query.h

```diff
@@ -494,6 +534,67 @@ class Query {
    */
   Status set_config(const Config& config);
 
+  /**
+   * Sets the data for a fixed/var-sized attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer The buffer that will hold the data to be read.
+   * @param buffer_size This initially contains the allocated
+   *     size of `buffer`, but after the termination of the function
+   *     it will contain the size of the useful (read) data in `buffer`.
+   * @param check_null_buffers If true (default), null buffers are not
+   * allowed.
+   * @return Status
+   */
+  Status set_data_buffer(
```

Comment:
we're relatively consistent using the `buffer_<postfix>` naming in the core. Rename this `set_buffer_data` and the like.

Reply:
I was just following similar convention with the new API `tiledb_query_set_[data/offset/validity]_buffer`. Will fix. Thanks